casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala

```diff
@@ -86,14 +101,46 @@ object EstimatorHelper {
       ancestors = maybeAncestors.flatten
       ancestorEvents = ancestors.flatMap(_.getBody.deploys.flatMap(_.deployLog)) ++
         ancestors.flatMap(_.getBody.deploys.flatMap(_.paymentLog))
-      ancestorChannels = ancestorEvents.flatMap {
-        case Event(Produce(produce: ProduceEvent)) =>
-          Set(produce.channelsHash)
-        case Event(Consume(consume: ConsumeEvent)) =>
-          consume.channelsHashes.toSet
-        case Event(Comm(CommEvent(Some(consume: ConsumeEvent), produces))) =>
-          consume.channelsHashes.toSet ++ produces.map(_.channelsHash).toSet
-        case _ => throw new RuntimeException("incorrect ancestor events")
-      }.toSet
+
+      produceEvents = ancestorEvents.map(extractProduce).flatten
+      consumeEvents = ancestorEvents.map(extractConsume).flatten
+      commEvents    = ancestorEvents.map(extractComm).flatten
+
+      producesInCommEvents = commEvents.flatMap(_.produces)
+      consumeInCommEvents  = commEvents.flatMap(_.consume)
```

Comment:
those two should not simply 'flatMap' - they should raise errors wherever the consume or the produces are empty

Reply:
@odeac

Reply:
the original implementation didn't fail on empty consumes or produces. I only suggested a way to write that such that it's more readable